### PR TITLE
AArch64: Implement Nestmate interface call

### PR DESCRIPTION
**Pull request by knn-k**
This commit implements Nestmate interface call for AArch64.

Signed-off-by: KONNO Kazuhiro <konno@jp.ibm.com>

### CODE REVIEW

**Comment by knn-k** [comment]
Based on the implementation for Power in #2809.

**Comment by knn-k** [comment]
jenkins test sanity alinux64 jdk11

**Comment by knn-k** [comment]
This change fixes the failures with `Nestmate_virtual_private_2` and `Nestmate_virtual_private_3`.
The test result is as expected.

**Comment by knn-k** [comment]
jenkins test sanity.system alinux64 jdk11

File: runtime/compiler/aarch64/runtime/PicBuilder.spp [hunk 1]
```diff
@@ -108,7 +109,12 @@
 	.set	J9TR_UICSnippet_CP,		8
 	.set	J9TR_UICSnippet_CPIndex,	16
 	.set	J9TR_ICSnippet_InterfaceClass,	24
-	.set	J9TR_ICSnippet_MethodIndex,	32
+	.set	J9TR_ICSnippet_ITableIndex,	32
+	.set	J9TR_ICSnippet_Class1,		40
+	.set	J9TR_ICSnippet_TargetAddr1,	48
+	.set	J9TR_ICSnippet_Class2,		56
+	.set	J9TR_ICSnippet_TargetAddr1,	64
```
Comment (0xdaryl) [inline runtime/compiler/aarch64/runtime/PicBuilder.spp]: This was already defined on L114.  Should be `J9TR_ICSnippet_TargetAddr2`?

File: runtime/compiler/aarch64/codegen/CallSnippet.cpp [hunk 2]
```diff
@@ -508,44 +508,97 @@ uint32_t TR::ARM64VirtualUnresolvedSnippet::getLength(int32_t estimatedSnippetSt
...
    *(intptrj_t *)cursor = 0;
-   cursor += 8;
+   cursor += sizeof(intptrj_t);
+
+   // Initialize for: two class ptrs, two target addrs
+   // Initialize target addrs with the address of the bl
+   *(intptrj_t *)cursor = -1;
+   *(intptrj_t *)(cursor+sizeof(intptrj_t)) = (intptrj_t)blAddress;
+   *(intptrj_t *)(cursor+2*sizeof(intptrj_t)) = -1;
+   *(intptrj_t *)(cursor+3*sizeof(intptrj_t)) = (intptrj_t)blAddress;
```
Comment (0xdaryl) [inline runtime/compiler/aarch64/codegen/CallSnippet.cpp]: What is the purpose of these fields?  Are they for building an interface PIC?  If I understand the code in this PR correctly, they are introduced but not used in this PR.  I would say your commit message should at least comment on the fact that you're introducing them here, but the actual PIC mechanics will be completed later.

Comment (knn-k) [reply runtime/compiler/aarch64/codegen/CallSnippet.cpp]: Updated the code removing the unused snippet fields.

**Comment by knn-k** [comment]
jenkins test sanity alinux64 jdk11

**Comment by knn-k** [comment]
Test result is as expected:
```
[2020-03-12T03:40:28.334Z] FAILED test targets:
[2020-03-12T03:40:28.334Z] 	cmdLineTester_pltest_0
[2020-03-12T03:40:28.334Z] 
[2020-03-12T03:40:28.334Z] TOTAL: 287   EXECUTED: 151   PASSED: 150   FAILED: 1   DISABLED: 6   SKIPPED: 130
```
Two `Nestmate_virtual_private_*` tests passed by applying this PR.
The failure with `cmdLineTester_pltest_0` is addressed by #8617 and #8711.

**Comment by 0xdaryl** [comment]
This will require a rebase now please.

**Comment by knn-k** [comment]
jenkins test sanity alinux64 jdk11

**Comment by knn-k** [comment]
Rebased, and also updated my code rewriting `intptrj_t` to `intptr_t` (#8835).

**Comment by knn-k** [comment]
The test result after rebase is the same as yesterday's result: Only `cmdLineTester_pltest_0` fails in sanity.functional.

**Comment by 0xdaryl** [comment]
This will need a PR for the 0.20.0 release as well please.

**Pull request merged**